component-monitor/src/main.rs

```diff
@@ -0,0 +1,238 @@
+use log::{debug, error, info, warn, LevelFilter};
+use nix::sys::inotify::{AddWatchFlags, InitFlags, Inotify, InotifyEvent};
+use signal_hook::{consts::SIGINT, consts::SIGTERM, iterator::Signals};
+use std::{fs, path::PathBuf, thread};
+use systemd_journal_logger::JournalLog;
+use walkdir::WalkDir;
+
+use clap::Parser;
+
+/// Manage the kernel-gpu-2404 contents
+#[derive(Debug, Parser)]
+#[command(version, about)]
+struct Cli {
+    /// The source path to monitor
+    #[arg(env = "COMPONENT_SOURCE")]
+    source: PathBuf,
+    /// The sentinel file name
+    #[arg(env = "COMPONENT_SENTINEL", value_parser = clap::builder::NonEmptyStringValueParser::new())]
```

Comment:
The sentinel here is actually a file name that we're monitoring for… I'm not reading in inotify events as arguments?

Or did I get my streams crossed here?

Reply:
_Oooooh,_ right, when I first read this, I thought `sentinel` here related to the inotify events! I think my confusion came from the use of the `String` type here---it looked like this specificially _isn't_ a file given that `target` has type `PathBuf` below. I think using `PathBuf` for both would clean this up nicely. 

For reference, `String` should be avoided for buffers as Rust's `String` and `&str` types both require valid UTF-8, so to drop this requrement, `PathBuf` is more idiomatic

Reply:
I suppose I thought that `Path*` would not be appropriate for just a file name… but of course a file name is still a path…

Reply:
Ah, but the string could still contain a `/` making it more like a path :) I think the intention is to use this with the file system, so signalling that could be benefical.

This actually makes me wonder, if `sentinel` is expected not to contain any path separators, rather than having separate target and sentinel parameters, could we take one and extract the name of the directory? Otherwise, I think `sentinel` feels more like a path to me

Reply:
> could we take one and extract the name of the directory?

I suppose if we define the source folder as the parent of the sentinel, that makes sense. And indeed avoids some rabbit holes.